lib/pbench/agent/tool_meister.py

```diff
@@ -1534,37 +1540,34 @@ def end_tools(self, data):
                 )
                 failures += 1
 
-        # Remove persistent tool temporary working directory on remote host
+        # Remove persistent tool temporary working directory
         directory = data["directory"]
         tool_dir = self.directories[directory]
 
-        if self._hostname == self._controller:
-            directory_to_delete = tool_dir
-        else:
-            directory_to_delete = tool_dir.parent
-
-        self.logger.info(
+        self.logger.debug(
             "%s: deleting persistent tool tmp directory %s %s",
             self._hostname,
             self._group,
-            directory_to_delete,
+            tool_dir,
         )
-        unexpected_files = [x for x in directory_to_delete.iterdir() if x.is_file()]
-        if not unexpected_files:
-            try:
-                shutil.rmtree(directory_to_delete)
-            except Exception:
-                self.logger.exception(
-                    f"Failed to remove tmp persistent tool data hierarchy, "
-                    f"{directory_to_delete}"
-                )
-                failures += 1
-        else:
+        unexpected_files = []
+        for dir, _, files in os.walk(tool_dir):
+            dirpath = Path(dir).relative_to(tool_dir)
+            if files:
+                unexpected_files += map(lambda x: f"{dirpath}/{x}", files)
```

Comment:
FWIW, you don't have to resort to using a `lambda` here -- `str.format()` is a callable which you can use `map()` to apply
```
                unexpected_files += map(f"{dirpath}/{{}}".format, files)
```
or you can just use a list comprehension:
```
                unexpected_files += [f"{dirpath}/{x}" for x in files]
```

Reply:
Ah! yeah I should have just used the list comprehension with f-strings. But I dont know if this change worth collapsing all the approvals.